libecole/src/reward/nnodes.cpp

```diff
@@ -0,0 +1,19 @@
+#include "ecole/reward/nnodes.hpp"
+
+#include "ecole/scip/model.hpp"
+
+namespace ecole {
+namespace reward {
+
+void NNodes::reset(scip::Model const&) {
+	last_n_nodes = 0;
+}
+
+Reward NNodes::obtain_reward(scip::Model const& model, bool /* done */) {
+	auto n_nodes_diff = SCIPgetNTotalNodes(model.get_scip_ptr()) - last_n_nodes;
```

Comment:
[SCIP Doc](https://www.scipopt.org/doc-6.0.2/html/group__PublicBranchingMethods.php#gab19f03a5d02f7dc16db722c862651933) states the function is only defined for stage `SCIP_STAGE_SOLVING`. What happens on terminal states? Or initial states (e.g. for use in `Configuring`)?

Reply:
The function is well defined in most stages. It is not well defined only if SCIP is in one of the following two stages:
SCIP_STAGE_INIT
SCIP_STAGE_FREE

https://www.scipopt.org/doc/html/group__PublicSolvingStatsMethods.php#ga0ab50fbdae44a36f3c7a5901ba20c71d

I have been playing with the reward function and Configuring, didn't have any issue.

Do we have any guarantee, for each environment, of the stages they might be in after `reset()` or `step()` ?

Reply:
Ho my bad, I linked against the wrong function.
I would say it is environment dependent. e.g. Configuring > reset> INIT, Branching > reset> SOLVING .